src/Symfony/Component/HttpKernel/Bundle/Bundle.php

```diff
@@ -149,4 +152,9 @@ private function parseClassName(): void
         $this->namespace = false === $pos ? '' : substr(static::class, 0, $pos);
         $this->name ??= false === $pos ? static::class : substr(static::class, $pos + 1);
     }
+
+    public function setContainer(?ContainerInterface $container): void
```

Comment:
should it be deprecated or is it meant to stay? same for the property?

Reply:
I would say it should stay. Some bundles are using this (like `FrameworkBundle`) in order to get container parameters in the `boot()` method:

https://github.com/symfony/symfony/blob/52a92926f7fed15cdff399c6921100a10e0d6f61/src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php#L105

Reply:
The `@method` annotation has been added to `BundleInterface`

Reply:
Actually, as BundleInterface extends the ContainerAwareInterface currently, the method is already required for implementations. So we can add it natively rather than through `@method`

Reply:
I agree, but isn't it more comfortable to have `@method` with `Not implementing it is deprecated since Symfony 6.4.` for the migration path?

But we could as well add this mention on top of the "native' method. I'll take care of removing it in https://github.com/symfony/symfony/pull/50578 (I couldn't find any example of this in the current codebase, only `@method` examples).

Reply:
it is **impossible** not to implement it in 6.4. As BundleInterface extends ContainerAwareInterface, the method is **already** part of the BundleInterface (as an inherited method).

that's not a case of adding a new method in BundleInterface but of moving it down